### PR TITLE
Fix filename bug

### LINKED ISSUE #53: Wrong GROMACS job file, if TPR file contains periods
Given a TPR file with the name `md_2016.4.tpr`, the command `mdbenchmark generate -n md_2016.4.tpr --module gromacs/2016.4` produces a wrong input file for the HPC job script.

**Expected behavior**

The `mpiexec` line should read:

```bash
mpiexec -n 24 -perhost 24 mdrun_mpi -v -maxh 0.25 -deffnm md_2016.4
```

**Actual behavior**

The line actually reads:

```bash
mpiexec -n 24 -perhost 24 mdrun_mpi -v -maxh 0.25 -deffnm md_2016.4.tpr
```

**Your version of MDBenchmark:**

mdbenchmark, 1.2.0

### PR DESCRIPTION
**Pull request by mimischi**
Fixes #53.

Changes made in this pull request:
- Change the way we handle filenames.

PR Checklist
------------
- [x] Added unit tests
- [x] Added changelog fragment in  `./changelog/` ([more information](https://github.com/bio-phys/MDBenchmark/blob/master/DEVELOPER.rst#using-towncrier))?
- [x] Issue raised/referenced?

### CODE REVIEW

File: mdbenchmark/generate.py [hunk 1]
```diff
@@ -126,8 +126,16 @@ def generate(name, gpu, module, host, min_nodes, max_nodes, time, list_hosts):
...
-            engine.write_bench(top, tmpl, n, gpu, m, engine_input, name, host,
-                               time)
+            engine.write_bench(
+                top=top,
+                tmpl=tmpl,
+                nodes=n,
+                gpu=gpu,
+                module=m,
+                full_filename=full_filename,
+                name=engine_filename,
```
Comment (kain88-de) [inline mdbenchmark/generate.py]: `name` isn't used anymore and it must have been defined.

**Comment by mimischi** [comment]
I cannot wrap my head around that. My local tests fail at [this line](https://github.com/bio-phys/MDBenchmark/blob/fix-filename/mdbenchmark/tests/mdengines/test_namd.py#L86), while Travis-CI does not seem to have any problems. I've cleared all caches already using `make clean`.

**Comment by mimischi** [comment]
Just realized what is causing the problem: for testing purposes I created `md.namd`, `md.psf` and `md.pdb` inside my `mdbenchmark` folder. Somehow `pytest` picks these up during testing and errors out. Deleting those resolves the issues.

Is something is wrong with our pytest setup?

**Comment by mimischi** [comment]
@kain88-de This is finished. I'd like to finish #49 before we release a new version.

**Comment by kain88-de** [comment]
> Somehow pytest picks these up during testing and errors out. Deleting those resolves the issues.

I don't know what files you had exactly so I can't say. To my knowledge, there isn't anything wrong. Also, test files should best be generated in a folder outside of the source code.

File: changelog/54.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+The provided filename is now correctly parsed and provided to the job templates.
```
Comment (kain88-de) [inline changelog/54.bugfix]: better 'Fixed edge-case in input filename parsing.'

It's not like it wasn't parsed correctly before. We just missed a specific edge case.

**Comment by mimischi** [comment]
> I don't know what files you had exactly so I can't say. To my knowledge, there isn't anything wrong. Also, test files should best be generated in a folder outside of the source code.

`touch md.namd`, `touch md.psf` and `touch md.pdb` inside the root folder. Then run `pytest` inside of that.

**Comment by kain88-de** [comment]
you invoke the CLI interface from the folder where you call pytest. To avoid this behavior you need to ensure that the cli invocation happens in a temporary folder.  It might be time to refactor the CLI testing functions.
